### PR TITLE
INDY-2003: add documentation for AUTH_RULE txn

### PR DESCRIPTION
**Pull request by Toktar**
Signed-off-by: toktar <renata.toktar@dsr-corporation.com>

### CODE REVIEW

**Comment by ashcherbakov** [comment]
(ci) test this please

File: docs/source/requests.md [hunk 1]
```diff
@@ -985,6 +985,209 @@ Command to change Pool's configuration
...
+   
+   New value that can be used to fill the field.
+
+**ConstraintType:**
+
+ConstraintList
+
+- `constraint_id` (enum: `AND` or `OR`):
+
+    Type of a constraint class. Needed for determine the type of constraint for correct deserialization.
```
Comment (ashcherbakov) [inline docs/source/requests.md]: It's needed to determine a type ...

File: docs/source/requests.md [hunk 2]
```diff
@@ -985,6 +985,209 @@ Command to change Pool's configuration
...
+  'auth_constraints': [<ConstraintEntity>,
+                      <ConstraintEntity>]
+}
+```
+    
+ConstraintEntity
+
+- `constraint_id` (enum: `ROLE`):
+
+   Type of constraint, which for ConstraintEntity is always of type "ROLE". Needed for determine the type of constraint for correct deserialization.
```
Comment (ashcherbakov) [inline docs/source/requests.md]: Type of a constraint. As of not only `ROLE` is supported, but plugins can register new ones.

Comment (ashcherbakov) [inline docs/source/requests.md]: It's needed to determine a type ...

File: docs/source/requests.md [hunk 3]
```diff
@@ -985,6 +985,209 @@ Command to change Pool's configuration
...
+    Role of a user that the NYM record is being created for. One of the following values
+
+    - None (common USER)
+    - 0 (TRUSTEE)
+    - 2 (STEWARD)
+    - 101 (TRUST_ANCHOR)
+    
+- `sig_count` (int):
+
+    The number of signatures that is needed to do some action described in the transaction fields.
```
Comment (ashcherbakov) [inline docs/source/requests.md]: ... to do the action ...

File: docs/source/requests.md [hunk 4]
```diff
@@ -985,6 +985,209 @@ Command to change Pool's configuration
...
+    
+- `need_to_be_owner` (boolean):
+
+    Flag to check is user must be owner of some transaction (Example: A steward must be the owner of the node to make changes to it).
```
Comment (ashcherbakov) [inline docs/source/requests.md]: .. if the user....

File: docs/source/requests.md [hunk 5]
```diff
@@ -985,6 +985,209 @@ Command to change Pool's configuration
...
+    
+- `metadata` (dict; optional):
+
+    Dictionary to additional parameters of constraint.
```
Comment (ashcherbakov) [inline docs/source/requests.md]: for additional parameters of the constraint. Can be used by plugins to add additional restrictions.

File: docs/source/requests.md [hunk 6]
```diff
@@ -985,6 +985,209 @@ Command to change Pool's configuration
...
+         'ver':'1',
+         'rootHash':'GJNfknLWDAb8R93cgAX3Bw6CYDo23HBhiwZnzb4fHtyi',
+         'auditPath':[  
+
+         ]
+      }
+   }
+```
+
+If format of transaction is incorrect, the client will received NACK message for request.
```
Comment (ashcherbakov) [inline docs/source/requests.md]: of a transaction

Comment (ashcherbakov) [inline docs/source/requests.md]: the client will receive .... for the request.

File: docs/source/requests.md [hunk 7]
```diff
@@ -985,6 +985,209 @@ Command to change Pool's configuration
...
+         'rootHash':'GJNfknLWDAb8R93cgAX3Bw6CYDo23HBhiwZnzb4fHtyi',
+         'auditPath':[  
+
+         ]
+      }
+   }
+```
+
+If format of transaction is incorrect, the client will received NACK message for request. 
+A client will receive NACK to
```
Comment (ashcherbakov) [inline docs/source/requests.md]: A client will receive NACK for

File: docs/source/requests.md [hunk 8]
```diff
@@ -987,12 +988,12 @@ Command to change Pool's configuration
...
 ### AUTH_RULE
 
-Command to change authentication rules. 
-Authentication rules are stored as key - value dictionary.
-Key - some action in the format `action--txn_type--field--old_value--new_value`
-Value is a set of constraints on the execution of this action. There are two types of constraints:
+A command to change authentication rules. 
+Authentication rules are stored as a key-value dictionary.
+A key is an authenticated action in the format `action--txn_type--field--old_value--new_value`.
+A value is a set of constraints on the execution of this action. The actions (keys) are static and can be found in [auth_rules.md](auth_rules.md). So, it's not possible to register new actions by this command. But it's possible to override authentication constraints (values) for a a given action. There are two types of constraints:
```
Comment (ashcherbakov) [inline docs/source/requests.md]: typo: `a a given`

File: docs/source/requests.md [hunk 9]
```diff
@@ -985,6 +986,211 @@ Command to change Pool's configuration
...
+  'auth_constraints': [<ConstraintEntity>,
+                      <ConstraintEntity>]
+}
+```
+    
+ConstraintEntity
+
+- `constraint_id` (enum: `ROLE`):
+
+   Type of a constraint. As of not only ROLE is supported, but plugins can register new ones. It's needed to determine a type of constraint for correct deserialization.
```
Comment (ashcherbakov) [inline docs/source/requests.md]: typo: `As of now`

File: docs/source/transactions.md [hunk 10]
```diff
@@ -709,6 +710,140 @@ Command to change Pool's configuration
...
+  'auth_constraints': [<ConstraintEntity>,
+                      <ConstraintEntity>]
+}
+```
+    
+AuthConstraint
+
+- `constraint_id` (enum: `ROLE`):
+
+      Type of a constraint. As of not only ROLE is supported, but plugins can register new ones. It's needed to determine a type of constraint for correct deserialization.
```
Comment (ashcherbakov) [inline docs/source/transactions.md]: typo: `As of now`

File: docs/source/transactions.md [hunk 11]
```diff
@@ -709,6 +710,140 @@ Command to change Pool's configuration
 }
 ```
 
+#### AUTH_RULE
+
+Transaction to change authentication rules. 
+Authentication rules are stored in the State as key - value dictionary. This coincides with the storage structure in (auth_map)[auth_rule.md]. 
+If config ledger does not have transaction for key, the State use value from auth_map.
```
Comment (ashcherbakov) [inline docs/source/transactions.md]: Better to say:
`If config ledger does not have a transaction for a given key, then a default rule defined in code is used.`

File: docs/source/requests.md [hunk 12]
```diff
@@ -985,6 +986,211 @@ Command to change Pool's configuration
...
+    - 2 (STEWARD)
+    - 101 (TRUST_ANCHOR)
+    
+- `sig_count` (int):
+
+    The number of signatures that is needed to do the action described in the transaction fields.
+    
+- `need_to_be_owner` (boolean):
+
+    Flag to check is the user must be owner of some transaction (Example: A steward must be the owner of the node to make changes to it).
```
Comment (ashcherbakov) [inline docs/source/requests.md]: if the user must be owner of a transaction

File: docs/source/transactions.md [hunk 13]
```diff
@@ -709,6 +710,140 @@ Command to change Pool's configuration
...
+    - 2 (STEWARD)
+    - 101 (TRUST_ANCHOR)
+   
+- `sig_count` (int):
+
+    The number of signatures that is needed to do the action described in the transaction fields.
+    
+- `need_to_be_owner` (boolean):
+
+    Flag to check is the user must be owner of some transaction (Example: A steward must be the owner of the node to make changes to it).
```
Comment (ashcherbakov) [inline docs/source/transactions.md]: if the user must be owner of a transaction

File: docs/source/requests.md [hunk 14]
```diff
@@ -985,6 +986,211 @@ Command to change Pool's configuration
...
+{
+    'sig_count': 1, 
+    'need_to_be_owner': False, 
+    'constraint_id': 'ROLE', 
+    'metadata': {}, 
+    'role': '0'
+}
+```
+
+*Request Example*:
```
Comment (Toktar) [inline docs/source/requests.md]: Fix incorrect json

**Pull request merged**